chain/x/zoracle/client/rest/types.go

```diff
@@ -17,22 +14,25 @@ type ScriptInfoWithTx struct {
 }
 
 type ReportDetail struct {
-	Reporter         sdk.ValAddress  `json:"reporter"`
-	TxHash           string          `json:"txhash"`
-	ReportedAtHeight int64           `json:"reportedAtHeight"`
-	ReportedAtTime   string          `json:"reportedAtTime"`
-	Value            json.RawMessage `json:"value"`
+	Reporter         sdk.ValAddress        `json:"reporter"`
+	TxHash           string                `json:"txhash"`
+	ReportedAtHeight int64                 `json:"reportedAtHeight"`
+	ReportedAtTime   string                `json:"reportedAtTime"`
+	Value            []types.RawDataReport `json:"value"`
 }
 
 type RequestQueryInfo struct {
-	ScriptInfo        types.ScriptInfo `json:"scriptInfo"`
-	CodeHash          cmn.HexBytes     `json:"codeHash"`
-	Params            json.RawMessage  `json:"params"`
-	TargetBlock       int64            `json:"targetBlock"`
-	Requester         sdk.AccAddress   `json:"requester"`
-	TxHash            string           `json:"txhash"`
-	RequestedAtHeight int64            `json:"requestedAtHeight"`
-	RequestedAtTime   string           `json:"requestedAtTime"`
-	Reports           []ReportDetail   `json:"reports"`
-	Result            json.RawMessage  `json:"result"`
+	OracleScriptID           int64                  `json:"oracleScriptID"`
+	Calldata                 []byte                 `json:"calldata"`
+	RequestedValidators      []sdk.ValAddress       `json:"requestedValidators"`
+	SufficientValidatorCount int64                  `json:"sufficientValidatorCount"`
+	ExpirationHeight         int64                  `json:"expirationHeight"`
+	IsResolved               bool                   `json:"isResolved"`
+	Requester                sdk.AccAddress         `json:"requester"`
+	TxHash                   string                 `json:"txhash"`
+	RequestedAtHeight        int64                  `json:"requestedAtHeight"`
+	RequestedAtTime          string                 `json:"requestedAtTime"`
```

Comment:
This can be put into its own struct if it helps with abstracting the transaction. The thing I mentioned above.